src/service/Offers.ts

```diff
@@ -41,33 +39,43 @@ export const getOfferService = (
       id,
       instanceHandle,
       publicInvitationMaker,
-      proposalTemplate: { give, want },
+      proposalTemplate: { give: giveTemplate, want: wantTemplate },
     } = offer;
 
-    const mapPursePetnamesToBrands = paymentProposals =>
-      Object.fromEntries(
+    const convertProposals = async paymentProposals => {
+      const entries = await Promise.all(
         Object.entries(paymentProposals).map(
           // @ts-expect-error
-          ([kw, { pursePetname, value }]) => {
-            const brand = pursePetnameToBrand.get(pursePetname);
-            if (!brand) {
+          async ([kw, { pursePetname, value, amount: serializedAmount }]) => {
+            /// XXX test e2e with dapp inter once feasible.
+            const amount = serializedAmount
+              ? await E(boardIdMarshaller).unserialize(serializedAmount)
+              : { brand: pursePetnameToBrand.get(pursePetname), value };
```

Comment:
we're cool with open-coding `AmountMath.make()`? maybe the notion of `brand` used here doesn't quite line up?

Reply:
Thanks, you caught a bug, I wasn't converting value to BigInt. Just changed to use AmountMath and confirmed it works.